runtime/rasdump/javadump.cpp

```diff
@@ -1484,6 +1484,93 @@ JavaCoreDumpWriter::writeEnvUserArgsHelper(J9VMInitArgs *vmArgs)
 				_OutputStream.writeCharacters("\n");
 			}
 		}
+
+		/* Ignored options for -Xjit and -Xaot follows the following rules:
+		 * 1. If Xjit is followed by -Xint or -Xnojit,
+			then any -Xjit:optionString beforehand appears as ignored
+		 * 2. If XX:+MergeCompilerOptions is present,
+			then all Xjit options after Xint or Xnojit should not be ignored
+		 * 3. If multiple Xjit string appear, all but the last one will appear as ignored
+		 *
+		 * and the same for -Xaot
+		 */
+
+		int numXjit = 0;
+		int numXaot = 0;
+		bool hasXXMerge = false;
+
+		/* These indices ensure all the options before them will be printed as ignored */
+		int ignoreXjitBefore = 0;
+		int ignoreXaotBefore = 0;
+
+		for (jint i = 0; i < args->nOptions; i++) {
+			const char * optionString = args->options[i].optionString;
+
+			if (0 == strncmp(optionString, "-Xint", 5)) {
+				/* Case 1: ignores all options before */
```

Comment:
5 must be 6 here - we don't want to match other options that start with `-Xint`. I think the same issue applies to `-Xnojit` `-Xnoaot`, `-XX:+MergeCompilerOptions` and `-XX:-MergeCompilerOptions`.

Please finish each sentence with a period.